File: game/src/main/java/net/demilich/metastone/game/entities/EntityType.java
package net.demilich.metastone.game.entities;

import co.paralleluniverse.common.util.Objects;

public enum EntityType {
	ANY,
	ACTOR,
	HERO,
	MINION,
	WEAPON,
	CARD,
	PLAYER,
	ENCHANTMENT,
	QUEST,
	SECRET;

	public boolean hasEntityType(EntityType other) {
		if (this == ANY) {
			return true;
		}

		if (this == ACTOR) {
			return other == HERO || other == MINION || other == WEAPON;
		}

		return Objects.equal(this, other);
	}
}

File: www/src/lib/metastone/EntityType.js
/**
 * Created by chris on 1/17/17.
 */
import EnumExtended from '../EnumExtended';

export default class EntityType extends EnumExtended {
}

EntityType.initEnum([
    'ANY',
    'HERO',
    'MINION',
    'WEAPON',
    'CARD',
    'PLAYER',
]);